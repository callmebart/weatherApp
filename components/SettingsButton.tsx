import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'


interface Props {
    onPress: () => void;
}

export default function SettingsButton({ onPress }: Props) {
    return (
        <View style={styles.modalButtonContainer}>
            <TouchableOpacity onPress={onPress} style={styles.settingsButton}>
                <Text style={styles.settingsButtonText}>Settings</Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    modalButtonContainer: {
        width: 281,
    },
    settingsButton: {
        width: 100,
        marginRight: 50,
        alignSelf: "flex-end",
        backgroundColor: "rgba(142,177,212,0.3)",
        justifyContent: "center",
        alignItems: "center",
        paddingVertical: 8,
        borderRadius: 8,
    },
    settingsButtonText: {
        color: "#404142",
        fontWeight: "bold",
    },
})