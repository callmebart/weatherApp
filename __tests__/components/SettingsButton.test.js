import { render } from "@testing-library/react-native";
import SettingsButton from "../../components/SettingsButton";

describe('SettingsButton', () => {
  it('should render', () => {
    const root  = render(<SettingsButton onPress={()=>null}/>).toJSON();
    expect(root).toMatchSnapshot();
  });
});