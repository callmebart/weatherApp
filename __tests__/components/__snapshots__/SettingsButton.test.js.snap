// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton should render 1`] = `
<View
  style={
    Object {
      "width": 281,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "backgroundColor": "rgba(142,177,212,0.3)",
        "borderRadius": 8,
        "justifyContent": "center",
        "marginRight": 50,
        "opacity": 1,
        "paddingVertical": 8,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#404142",
          "fontWeight": "bold",
        }
      }
    >
      Settings
    </Text>
  </View>
</View>
`;
